### DISCUSSION TITLE
Is it possible to integrate Apple Pay using swell-js?

### DISCUSSION THREAD

**Discussion by Buupu** [swellstores/community#130] [Help] [answered]
I've had a look at the docs and can't find any information on this.

Is it possible to setup apple pay as a payment method with swell-js?

**Comment by cristianstanescu** [comment]
@Buupu you should be able to integrate Apple Pay using swell-js via Braintree. While we don't have a specific code example, hopefully these two can get you started:

- [Activate Apple Pay with Braintree](https://www.swell.is/help/store-setup/payment-gateways#activate-apple-pay-with-braintree)
- [PayPal via Braintree](https://developers.swell.is/frontend-api/payments#paypal-via-braintree)

Please let us know how it goes!

**Comment by swellmike** [comment]
FWIW, the Swell team is actively working on an Apple & Google Pay integration via Braintree/Stripe for our hosted checkout.  Eventually some of that code should find its way to our public SDKs (e.g., swell-js).

**Comment by sags95** [comment]
@Buupu currently swell-js only renders Stripe's Card Element. In the meantime, if you wanted to use Stripe's Payment Request Element for Apple/Google Pay you might be interested in my response here: https://github.com/orgs/swellstores/discussions/116. Otherwise if you're using Braintree then it would be possible to also use their Drop-In UI and pass the `nonce` to the cart.

**Comment by swellmike** [comment, marked as answer]
swell-js v3.20.0 now has support for Apple Pay & Google Pay via Braintree / Stripe.

**Reply by Buupu** [reply]
Awesome! Excited to use it.
